carbon-design-system/carbon
# fix(date-picker): clearing input and selecting a date

Closes #14529

When opening the date-picker, then clearing the input in date-picker and then clicking on a value the date-picker would still show empty. 

https://github.com/user-attachments/assets/11c35e5d-81b1-4178-b050-29791537fc10


https://github.com/user-attachments/assets/084f4c4f-7f58-4e17-82be-0f1371a27786


#### Changelog

**Changed**

- updated handleOnChange and added a keyPress event listener

#### Testing / Reviewing

make sure tests still pass. make sure works similiarly to flatpickr examples

Open storybook
- go to date-picker stories
- open the date picker select a date
- open again and clear the current date
- while it is still open click on a date
- date-picker should hold that current value in the input
